mne/time_frequency/tfr.py

```diff
@@ -2173,13 +2189,17 @@ def combine_tfr(all_tfr, weights='nave'):
 
 
 def _get_data(inst, return_itc):
-    """Get data from Epochs or Evoked instance as epochs x ch x time."""
+    """Get data from Epochs, Evoked, or Raw instance as epochs x ch x time."""
     from ..epochs import BaseEpochs
     from ..evoked import Evoked
-    if not isinstance(inst, (BaseEpochs, Evoked)):
-        raise TypeError('inst must be Epochs or Evoked')
+    from ..io import BaseRaw
+    if not isinstance(inst, (BaseEpochs, Evoked, BaseRaw)):
+        raise TypeError('inst must be Epochs or Evoked or Raw')
     if isinstance(inst, BaseEpochs):
         data = inst.get_data()
+    elif isinstance(inst, BaseRaw):
+        # make data into 3D to play nice with epochs x ch x time
```

Comment:
should you also do this:

https://github.com/mne-tools/mne-python/blob/8bc55d7435ce047c37465f2b5c173517491d7002/mne/time_frequency/tfr.py#L2204

I am thinking that maybe you can use the same code path for Evoked and Raw? They basically both contain 2D data arrays

Reply:
Like what's done here: c14d080?